internal/app/migration.go

```diff
@@ -0,0 +1,52 @@
+package app
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"go.uber.org/zap"
+)
+
+func migrationRun(dsn string, log *zap.Logger) error {
+	db, err := sql.Open("postgres", dsn)
+	if err != nil {
+		return err
+	}
+
+	driver, err := postgres.WithInstance(db, &postgres.Config{})
+	if err != nil {
+		return err
+	}
+
+	sourceURL := "file://migrations"
+	var pwd string
+	if pwd, err = os.Getwd(); err == nil {
```

Comment:
This is a good example why you need to use `embbed`.

Reply:
fixed